dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/cache/impl/TaskInstanceCacheManagerImpl.java

```diff
@@ -110,6 +110,7 @@ public void cacheTaskInstance(TaskExecutionContext taskExecutionContext) {
     @Override
     public void cacheTaskInstance(TaskExecuteAckCommand taskAckCommand) {
         TaskInstance taskInstance = new TaskInstance();
+        taskInstance.setId(taskAckCommand.getTaskInstanceId());
```

Comment:
There is another problem that you can fix together.
This method is used to update the task instance status and set some attribute, in the line 119, it's better to update the taskInstance if the taskInstance is already exist in cache rather than replace.

Reply:
@wordhardqi Since we use ConcurrentHashMap here, can we direct use `computeIfPresent` method here?

This method `cacheTaskInstance` is used to update status from worker response, and the task instance should already be cached in the map, so we don't need to considerate the absent case.